price_list.py

```diff
@@ -1,16 +1,30 @@
 from trytond.pool import PoolMeta
 from trytond.model import fields
 from trytond.pyson import Bool, Eval
+from trytond.modules.product import round_price
+from .discount import ApplyDiscountMixin
 
 
-class PriceList(metaclass=PoolMeta):
+class PriceList(ApplyDiscountMixin, metaclass=PoolMeta):
     __name__ = 'product.price_list'
 
     def compute_discount_formula(self, product, quantity, uom, pattern=None):
         line = self.get_price_line(product, quantity, uom, pattern=pattern)
         if line:
             return line.discount_formula
 
+    def compute(self, product, quantity, uom, pattern=None):
+        unit_price = super().compute(product, quantity, uom, pattern)
+
+        line = self.get_price_line(product, quantity, product.default_uom,
+            pattern=pattern)
+        if line and line.discount_formula is not None:
```

Comment:
This is not correcte, I think.

If there's a discount_formula, then formula must be zero. So you have to pick base_price and apply the discount on top of base_price, not unit_price.

Reply:
formula must be zero,[ in sale_discount_price_list  module](https://github.com/NaN-tic/trytond-sale_discount_price_list/blob/main/price_list.py#L83)

Reply:
```
if line and line.discount_formula is not None:
    base_price = self.compute_base_price(
                product, quantity, product.default_uom, pattern=pattern)
    if base_price is not None:
         unit_price = self.apply_discount_formula(base_price,
                line.discount_formula, raise_exception=False)
```

Reply:
436acf2fed19ac4c3b72485ee48f5ec644a3030c